src/web/components/SharingPermission/SharingPermissionsTable.tsx

```diff
@@ -48,71 +49,96 @@ export function SharingPermissionsTable({
     return sharingParticipants.filter((p) => checkedParticipants.has(p.siteId!));
   }, [checkedParticipants, sharingParticipants]);
 
-  const tableHeader = () =>
-    !hasParticipantSelected ? (
-      <>
-        <SortableTableHeader<AvailableParticipantDTO> sortKey='name' header='Participant Name' />
-        <th>Participant Type</th>
-        <th>Added By</th>
-      </>
-    ) : (
-      <th colSpan={3}>
-        <Dialog
-          title='Are you sure you want to delete these permissions?'
-          triggerButton={
-            <button className='transparent-button sharing-permission-delete-button' type='button'>
-              <FontAwesomeIcon
-                icon={['far', 'trash-can']}
-                className='sharing-permission-trashcan-icon'
-              />
-              Delete Permissions
-            </button>
-          }
-          open={openConfirmation}
-          onOpenChange={setOpenConfirmation}
+  const handleCheckboxChange = () => {
+    if (selectAllState === TriStateCheckboxState.unchecked) {
+      setCheckedParticipants(new Set(filteredParticipants.map((p) => p.siteId!)));
+    } else {
+      setCheckedParticipants(new Set());
+    }
+  };
+
+  const isSelectedAll = useMemo(() => {
+    if (!filteredParticipants.length) return false;
+    return filteredParticipants.every((p) => checkedParticipants.has(p.siteId!));
+  }, [filteredParticipants, checkedParticipants]);
+
+  useEffect(() => {
+    if (isSelectedAll) {
+      setSelectAllState(TriStateCheckboxState.checked);
+    } else if (checkedParticipants.size > 0) {
+      setSelectAllState(TriStateCheckboxState.indeterminate as CheckedState);
+    } else {
+      setSelectAllState(TriStateCheckboxState.unchecked);
+    }
+  }, [checkedParticipants.size, isSelectedAll]);
+
+  const deletePermissionBtn = (
+    <Dialog
+      title='Are you sure you want to delete these permissions?'
+      triggerButton={
+        <button className='transparent-button sharing-permission-delete-button' type='button'>
+          <FontAwesomeIcon
+            icon={['far', 'trash-can']}
+            className='sharing-permission-trashcan-icon'
+          />
+          Delete Permissions({selectedParticipantList.length})
+        </button>
+      }
+      open={openConfirmation}
+      onOpenChange={setOpenConfirmation}
+    >
+      <div className='dialog-body-section'>
+        <ul className='dot-list'>
+          {selectedParticipantList.map((participant) => (
+            <li key={participant.id}>{participant.name}</li>
+          ))}
+        </ul>
+        <p>Note: Sharing will continue with participants that are shared via &quot;Auto&quot;.</p>
+      </div>
+      <div className='dialog-footer-section'>
+        <button type='button' className='primary-button' onClick={handleDeletePermissions}>
+          I want to Remove Permissions
```

Comment:
I think this copy is just "Delete Permissions" now 
![image](https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/62bbafb9-60f9-4e8c-9cba-52403a47ce1d)

Reply:
Thanks, You're right, I didn't check the copy change. Done